databricks/koalas/indexes.py

```diff
@@ -587,6 +587,66 @@ def copy(self, name=None):
             result.name = name
         return result
 
+    def droplevel(self, level):
+        """
+        Return index with requested level(s) removed.
+        If resulting index has only 1 level left, the result will be
+        of Index type, not MultiIndex.
+
+        Parameters
+        ----------
+        level : int, str, or list-like, default 0
+            If a string is given, must be the name of a level
+            If list-like, elements must be names or indexes of levels.
+
+        Returns
+        -------
+        Index or MultiIndex
+
+
+        Examples
+        --------
+        >>> midx = ks.DataFrame({'a': ['a', 'b']}, index=[['a', 'x'], ['b', 'y'], [1, 2]]).index
+        >>> midx  # doctest: +SKIP
+        MultiIndex([('a', 'b', 1),
+                    ('x', 'y', 2),
+                   )
+
+        >>> midx.droplevel([0, 1])
+        Int64Index([1, 2], dtype='int64')
+
+        >>> midx.droplevel(0) # doctest: +SKIP
+        MultiIndex([('b', 1),
+                    ('y', 2)]
+                   )
+        """
+        if not isinstance(level, (tuple, list)):
+            level = [level]
+
+        for l in level:
+            if isinstance(l, int) and (l > self.nlevels - 1):
+                raise IndexError("Too many levels: Index has only {} levels, not {}"
+                                 .format(self.nlevels, l+1))
+            if isinstance(l, str) and (l not in self._internal.index_columns):
+                raise KeyError("Level {} not found".format(l))
+
+        if len(level) >= self.nlevels:
+            raise ValueError(
+                "Cannot remove {} levels from an index with {} "
+                "levels: at least one level must be "
+                "left.".format(len(level), self.nlevels)
+            )
+
+        index_columns = [v for k, v in enumerate(self._internal.index_columns)
+                         if ((k not in level) & (v not in level))]
+        internal = _InternalFrame(sdf=self._internal._sdf.select(index_columns),
+                                  index_map=[(i, None) for i in index_columns])
```

Comment:
We should preserve the index names if the index are named.

```py
>>> midx
MultiIndex(levels=[['a', 'x'], ['b', 'y'], [1, 2]],
           codes=[[0, 1], [0, 1], [0, 1]],
           names=['l1', 'l2', 'l3'])
>>> midx.droplevel(0)
MultiIndex(levels=[['b', 'y'], [1, 2]],
           codes=[[0, 1], [0, 1]])
```

whereas

```py
>>> midx.to_pandas().droplevel(0)
MultiIndex(levels=[['b', 'y'], [1, 2]],
           codes=[[0, 1], [0, 1]],
           names=['l2', 'l3'])
```

Reply:
ah sorry, @RainFung 

what i meant at https://github.com/databricks/koalas/pull/1091#discussion_r352353249,

we should consider the both case when index has its own name or not.

your existing code in previous comment could handle only the case index has own name.

so we should consider both case.